File: fibonacci/Fibonacci.java
/**
 * Created by daniel on 2/28/16.
 */
public class Fibonacci {
    private int max;
    private int[] fibs;

    public Fibonacci (int n) {
        max = n;
        fibs = new int[max + 1];
        fibs[0] = 0;
        fibs[1] = 1;
        for (int i = 2; i <= max; i++) {
            fibs[i] = fibs[i - 1] + fibs[i - 2];
        }
    }

    public String toString () {
        StringBuilder out = new StringBuilder();
        out.append(fibs[0]);

        for (int i = 1; i <= max; i++) {
            out.append("\n");
            out.append(fibs[i]);
        }
        return out.toString();
    }

    public static void main (String[] args) {
        int parsedArg = Integer.parseInt(args[0]);

        Fibonacci fibonacci = new Fibonacci(parsedArg);
        System.out.println(fibonacci.toString());
    }
}

File: Fibonacci/Fibonacci.scala
/**
 * Fibonacci.scala
 * by Daniel Brice
 * Invoke as `scala Fibonacci <num>`
 * Prints the first <num> Fibonacci numbers, one per line
 */

object Fibonacci {

  def genFibs(n: Int): List[Int] = {

    var fibs = List(1, 0)

    while(fibs.length < n + 1) {
      fibs = fibs.head + fibs.tail.head :: fibs
    }

    fibs
  }

  def main(args: Array[String]): Unit = {
    val fibs = genFibs(args(0).toInt).reverse
    fibs.map(println)
  }
}
